pyiron_base/database/filetable.py

```diff
@@ -56,9 +56,20 @@ class FileTableSingleton(ABCMeta):
     _instances = {}
 
     def __call__(cls, index_from):
-        _path = os.path.abspath(index_from)
+        _path = os.path.abspath(os.path.expanduser(index_from))
         if _path not in cls._instances:
-            cls._instances[_path] = super(FileTableSingleton, cls).__call__(index_from)
+            common_path = get_most_common_path(
+                path=_path, reference_paths=cls._instances.keys()
+            )
+            if common_path is not None:
```

Comment:
I see that this was extracted to a method then that change was reverted. I can live with it as is -- although I still prefer to avoid the nested if-else by pulling it out -- but I'm curious why it got reverted? Does it not work?

Reply:
I got a strange error calling `__call__()`: 
```
  File "/home/runner/work/pyiron_base/pyiron_base/pyiron_base/database/filetable.py", line 69, in _get_common_fileindex
    return super(FileTableSingleton, cls).__call__(index_from=path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: descriptor '__call__' of 'type' object needs an argument
```

Reply:
Cool, no worries then. Thanks for the trace too, I'll experiment with a MWE to see if I can figure out the python of what's going on here, just for educational purposes.